Core/clim-basic/decls.lisp

```diff
@@ -196,6 +196,22 @@
 (defgeneric sheet-occluding-sheets (sheet child))
 (defgeneric map-over-sheets (function sheet))
 
+(defgeneric clim-extensions:sheet-name (sheet)
+  (:documentation "McCLIM extension: Return the name of SHEET.
+The returned name is a symbol and does not change.
+Only top-level sheets have a name. For sheets which are also panes, the
```

Comment:
"Only top-level sheets have a name" - top-level-sheet-pane is not an exported symbol, so it doesn't make sense to mention it here.  Maybe there should be a default method resolving to `unnamed-sheet` (name) and `"(Unnamed Sheet)"`? Also we may want to export named-sheet-mixin in clime.

Other than that, I think we should keep package prefix out of the definitions. All definitions reside in climi package, so there should be no conflicts and that way we'll be able to just skip the symbol from clime exports to change its home package.